DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/data_bundle/src/data_bundle.cpp

```diff
@@ -0,0 +1,187 @@
+#include <iomanip>
+#include <iostream>
+#include <numeric>
+#include <vector>
+
+#include <CL/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+
+// dpc_common.hpp can be found in the dev-utilities include folder.
+// e.g., $ONEAPI_ROOT/dev-utilities//include/dpc_common.hpp
+#include "dpc_common.hpp"
+
+#include "memory_utils.hpp"
+#include "metaprogramming_utils.hpp"
+#include "unrolled_loop.hpp"
+
+using namespace sycl;
+
+////////////////////////////////////////////////////////////////////////////////
+template <typename T, int n>
+struct DataBundle {
```

Comment:
Why not put DataBundle in a header file in the include directory?

Reply:
We could, but I want the idea to be that the memory utilities work with any bundle-esque class, where DataBundle is one example.

Reply:
I like the plan to make the memory utilities generic, but I still think providing this bundle class in a separate header would be useful.